### DISCUSSION TITLE
Can't install Debian when also specifying --no-launch

### DISCUSSION THREAD

**Discussion by adras** [microsoft/WSL#10081] [General]
### Windows Version

doesn't matter, Windows 10, Windows 11

### WSL Version

1.2.5.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [X] WSL 1

### Repro Steps

wsl --unregister Debian
wsl --install Debian --web-download -n
wsl -l

### Expected Behavior

wsl should install Debian but not start it

### Actual Behavior

wsl states that Debian is already installed.
wsl -l states that Debian is not installed

**Comment by elsaco** [comment]
@adras in step 2 you're trying to download Debian and the app is telling the package is already on your system. Use `winget` or `get-appxpackage` to check. If you run `wsl --install` it will register the distro with WSL but not download it if locally present. When you run `wsl --unregister` it deletes the distro data not the distro package.

```
PS C:\Users\elsaco> wsl --install Debian --web-download -n
Downloading: Debian GNU/Linux
Installing: Debian GNU/Linux
Debian GNU/Linux has been installed.
```

however WSL won't see it until the setup is complete:

```
PS C:\Users\elsaco> wsl -d Debian
There is no distribution with the supplied name.
Error code: Wsl/Service/WSL_E_DISTRO_NOT_FOUND
PS C:\Users\elsaco> wsl --install Debian
Debian GNU/Linux is already installed.
Launching Debian GNU/Linux...
Installing, this may take a few minutes...
Please create a default UNIX user account. The username does not need to match your Windows username.
For more information visit: https://aka.ms/wslusers
Enter new UNIX username:
```

**Comment by adras** [comment]
This is very confusing. What I'd like to do is to setup my distribution from a batch file, similar to a container. However, when I execute `wsl --install Debian` like in your example, It's getting started right away, asking for username and password. Is there a way to prevent that? I just need it installed, so I can run it later

**Comment by NotTheDr01ds** [comment]
Yes, I agree it's confusing, but you'll get it ;-).  Let me rephrase what @elsaco said:

There are two different *parts* to the Debian installation:

* The "distribution"* (what you see with `wsl -l -v`)
* The application package, which contains the rootfs and the "distro launcher" which sets things up for you on initial launch.
 
When you are testing your installation process that you describe, you need to remove *both*:

* To remove the Debian distribution, use `wsl --unregister debian` as you've been doing.
* The application package can be removed using any of the following methods:

    * Searching for "Debian" in the Start menu, right-clicking it, and selecting *"Uninstall"*
    * Searching for "Debian" in the *"Add or remove programs"* Settings page and selecting *"Uninstall"* from the "3 dots" menu
    * From PowerShell, using `Get-AppxPackage -Name "TheDebianProject.DebianGNULinux" | Remove-AppxPackage`

After removing *both* distribution and application package, then running:

```
wsl --install Debian --web-download -n
```

... will re-download the application package to your system.  You can then finalize the distribution installation using:

* "Debian" on the Start menu
* Or running `debian.exe` from PowerShell
